app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -111,15 +113,26 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
             )
         }
 
-        if (Settings.getInstance(this@MainActivity).shouldShowPocketOnboarding()) {
-            val onboardingIntents =
-                    Intent(this@MainActivity, PocketOnboardingActivity::class.java)
-            startActivity(onboardingIntents)
-        }
-
         if (Settings.getInstance(this@MainActivity).shouldShowTurboModeOnboarding()) {
             val onboardingIntent = Intent(this@MainActivity, OnboardingActivity::class.java)
-            startActivity(onboardingIntent)
+            startActivityForResult(onboardingIntent, ONBOARDING_REQ_CODE)
+        }
+    }
+
+    // TODO: this is hot fix until NavigationOverlayFragment refactoring is done; we should use LiveData instead
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == ONBOARDING_REQ_CODE && resultCode == Activity.RESULT_OK) {
+            val fragmentManager = supportFragmentManager
+            val webRenderFragment = fragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG) as WebRenderFragment?
+
+            webRenderFragment?.updateOverlayTurboMode()
+
+            if (Settings.getInstance(this@MainActivity).shouldShowPocketOnboarding()) {
```

Comment:
This is a workaround, for startActivityForResult being incredibly slow to finish when there is another activity behind. We won't need to worry about this once we can use LiveData for this, so this workaround for now is fine.